pkgs/test/nixpkgs-check-by-name/workflows/check-by-name/reproduce.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -e
+# Usage: pkgs/test/nixpkgs-check-by-name/reproduce.sh <base SHA> <tooling SHA> <merged SHA>
+
+# TODO(amjoseph): allow omitting the final argument, since it is
+# often a commit which exists only in github (and is difficult to
+# `git fetch` from github before the PR is merged).
```

Comment:
Ideally we'd have two scripts:
- `./reproduce.sh <tooling SHA> <base SHA> <merged SHA>`: Reproducible from the args as here (I'd order the args this way).
- `./reproduce-latest.sh <base branch name>`: Tests the current HEAD against the latest commit of the base branch name, effectively replicating what a new CI run should produce. This should first determine the arguments for `./reproduce.sh` and then call it.

Can you implement something like that?